pkg/ioutils/writeflusher.go

```diff
@@ -1,32 +1,48 @@
 package ioutils
 
 import (
+	"fmt"
 	"io"
 	"net/http"
 	"sync"
 )
 
-// WriteFlusher wraps the Write and Flush operation.
+// WriteFlusher wraps the Write and Flush operation. In addition, the Close
+// method can be called to intercept Read/Write calls if the targets lifecycle
+// has already ended.
 type WriteFlusher struct {
-	sync.Mutex
+	mu      sync.Mutex
 	w       io.Writer
 	flusher http.Flusher
 	flushed bool
+	closed  error
```

Comment:
Allows easy future extension for `CloseWithErr`. It is also arguable that we may want to pick up the actual write error and tombstone the flusher.